### PR TITLE
RUMM-1942 Improve the perfs measurements methods in nightly tests

### PR DESCRIPTION
**Pull request by mariusc83**
### What does this PR do?

In some of our perfs measurement methods we were also taking int account the time to forge the fake method parameters. 
This PR fixes this problem by generating all the parameters outside of the measurement block.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/rum/RumMonitorBackgroundE2ETests.kt [hunk 1]
```diff
@@ -103,12 +104,13 @@ class RumMonitorBackgroundE2ETests {
...
+        )
         measure(testMethodName) {
             GlobalRum.get().addUserAction(
-                forge.aValueFrom(
-                    RumActionType::class.java,
-                    exclude = listOf(RumActionType.CUSTOM)
-                ),
+                actionType,
                 actionName,
                 attributes = defaultTestAttributes(testMethodName)
```
Comment (0xnm) [inline instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/rum/RumMonitorBackgroundE2ETests.kt:115]: probably we should also extract `attributes` here as it was done in other places

File: instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/rum/RumMonitorBackgroundE2ETests.kt [hunk 2]
```diff
@@ -194,19 +200,21 @@ class RumMonitorBackgroundE2ETests {
...
             resourceKey,
-            attributes = defaultTestAttributes(testMethodName)
+            attributes = attributes
         )
         measure(testMethodName) {
             GlobalRum.get().stopResource(
                 resourceKey,
                 200,
                 forge.aLong(min = 1),
                 forge.aValueFrom(RumResourceKind::class.java),
```
Comment (0xnm) [inline instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/rum/RumMonitorBackgroundE2ETests.kt:216]: should extract those

File: instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/rum/RumMonitorBackgroundE2ETests.kt [hunk 3]
```diff
@@ -194,19 +200,21 @@ class RumMonitorBackgroundE2ETests {
     fun rum_rummonitor_stop_background_resource() {
         val testMethodName = "rum_rummonitor_stop_background_resource"
         val resourceKey = forge.aResourceKey()
+        val method = forge.aResourceMethod()
+        val attributes = defaultTestAttributes(testMethodName)
         GlobalRum.get().startResource(
             resourceKey,
-            forge.aResourceMethod(),
```
Comment (0xnm) [inline instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/rum/RumMonitorBackgroundE2ETests.kt:199]: minor: this call is not in the `measure` block, so no necessary to extract

**Review by 0xnm — CHANGES_REQUESTED** [review]
Nice idea! I commented few places for the improvement, otherwise lgtm.

File: instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/rum/RumMonitorBackgroundE2ETests.kt (hunk 1 above)
Comment (mariusc83) [reply instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/rum/RumMonitorBackgroundE2ETests.kt:115]: Good catch

**Review by 0xnm — APPROVED** [review]

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -315,5 +315,9 @@ fun $EXP$(){
 
 Because of our crash handling [test cases](instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/crash), when running the nightly tests through gradle,  
 the process does not finish properly so it may happen that will keep hanging until it will eventually timeout.To avoid this issue you should run the tests through adb shell directly 
-using the instrumentation tool: `adb shell am instrumentation -w -e package com.datadog.android.nightly.[package] com.datadog.android.nightly.test/androidx.test.runner.AndroidJUnitRunner`
+using the instrumentation tool: 
+```
+./gradlew :instrumented:nightly-tests:installDebug && ./gradlew :instrumented:nightly-tests:installDebugAndroidTest &&
+adb shell am instrument -w -e package com.datadog.android.nightly.[package] com.datadog.android.nightly.test/androidx.test.runner.AndroidJUnitRunner
```
[the remark this answers is not in the document]
Comment (xgouchet) [reply CONTRIBUTING.md:321]: Can you also explain what `[package]` is supposed to be ?

**Review by xgouchet — APPROVED** [review]

**Comment by xgouchet** [comment]
We might want to also add a detekt / static analysis check to ensure we find everywhere this antipattern is used

**Comment by mariusc83** [comment]
> We might want to also add a detekt / static analysis check to ensure we find everywhere this antipattern is used

Yup, I will add a JIRA task for this.

**Review by xgouchet — APPROVED** [review]

**Pull request merged**